12-deploy-python-microservice.ipynb

```diff
@@ -0,0 +1,730 @@
+   "execution_count": null,
+   "id": "6668f639",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "light_data['areaServed'] = [{\"type\": \"Point\", \"coordinates\": coordinates} for coordinates in all_coordinates[1:]]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4628e7b5",
+   "metadata": {},
+   "source": [
+    "* Finally, we make the request updating your Smart Village deployment with the coordinate data.\n",
+    "We can print as much as debug information as we need. In this care we have 3 print statements. In the function these can be combined or modified as you see fit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "00d8866e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if map_server is not None:\n",
+    "    put_response = requests.put(url=map_server+smart_traffic_light_api, json={\"list\":[light_data]}, headers=headers)\n",
+    "    if put_response.status_code == 200:\n",
+    "        put_data = put_response.json()\n",
+    "        print(f\"Successful PUT, Updated: {put_data}\")\n",
+    "    else:\n",
+    "        print(f\"PUT Failed: {put_response.status_code}, {put_response.text}\")\n",
+    "else:\n",
+    "    print(\"No MAP_SERVER_URL environment variable defined, printing payload to terminal:\")\n",
+    "    print(f\"My light_data is: {light_data}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fd76a33c",
+   "metadata": {},
+   "source": [
+    "#### Setting up the service code\n",
+    "The `@app.route('/health')` annotation on top of a function is what tells flask what function to call when we access that http endpoint.\n",
+    "The `check_data` function will call the `get_map_data` function and then update your Smart Village deployment with that data by calling `update_data`. \n",
+    "These are the functions you must create from the above sections."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "520a0504",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@app.route('/health')\n",
+    "def check_data():\n",
+    "    map_data = get_map_data()\n",
+    "    update_data(map_data)\n",
+    "    return \"App is running\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c0c7a7db",
+   "metadata": {},
+   "source": [
+    "To open the port used for talking to the service we need to define in code or at runtime. \n",
+    "For this instance we are going to programmatically setup the host/port that way we can easily import this application to OpenShift using source to image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "919c6235",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if __name__ == '__main__':\n",
+    "     app.run(host='0.0.0.0', port=8080, debug=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1f96ef6a",
+   "metadata": {},
+   "source": [
+    "## Building a container with my application.\n",
```

Comment:
At this point, we'll probably need to explain how to git add/commit/push their code up to a new repo in their Github with a personal access token so that they can build the container on their provided laptop and not in the cloud anymore: 

Click [here to generate a GitHub personal access token](https://github.com/settings/tokens?type=beta). Make sure you are creating a  fine-grained access token which can grant write access to your personal repositories. 

Give your token a note that you can remember. 

give your token access to clone your repository. 

You will want to remember the random generated token value for later. You can copy it to the clipboard with the button on the right. 

```bash
git config --global user.email "MyEmail@example.com"
git config --global user.name "FirstName LastName"
git config --global credential.helper store
```